create-plugins-list-file.sh

```
@@ -36,6 +36,30 @@ get_plugin_name() {
     echo "$url" | sed -n 's#.*/\([^/]*\)$#\1#p'
 }
 
+# Function to extract Java version from pom.xml
+get_java_version_from_pom() {
+    local pom_file=$1
+    local java_version=$(xmllint --xpath "string(//properties/maven.compiler.source)" "$pom_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)